### ISSUE TITLE
Use `tokio_sink::mpsc` over `futures::sync::mpsc`

### ISSUE THREAD

**Issue by LucioFranco** [timberio/vector#579]
`tokio-sync` has a more performant channel that we may want to consider moving too. This change should be benchmarked and proven that it does indeed give us the perf increase.

cc @lukesteensen

**Comment by LucioFranco** [comment]
Related #2184

**Comment by binarylogic** [comment]
Noting, this will be covered in https://github.com/timberio/vector/issues/2625

**Comment by lukesteensen** [comment]
This is blocked on #2935

**Comment by kit70669** [comment]
Can someone give me info on all this I can donate

**Comment by lukesteensen** [comment]
@kit70669 Thank you for the offer! I actually believe this will tie into some very core systems of Vector, so it is probably best for a core team member to tackle this particular issue. I am actually working towards solving it right now.

**Comment by lukesteensen** [comment]
Closing in favor of the four more specific issues referenced above.
